nucypher/policy/models.py

```diff
@@ -26,11 +26,12 @@ class Arrangement:
     A Policy must be implemented by arrangements with n Ursulas.  This class tracks the status of that implementation.
     """
     federated = True
+    ID_LENGTH = 32
```

Comment:
Is there a particular reason to have arrangement IDs 32 bytes long?

Reply:
I guess this depends only in the UX treatment of arrangement IDs. If in the end they're only used internally in our protocols, then 32 bytes it's fine; if, otherwise, users need to interact with them, then 32-bytes arrangement IDs can be too big.

Reply:
Can you point me to a simple, quick document that guides the selection of choosing minimum viable sizes of bytestrings via different means so as to preclude collision?

Reply:
If the only concern is avoiding collisions, and given IDs are uniformly sampled, then we can reuse the probability tables from [birthday attacks](https://en.wikipedia.org/wiki/Birthday_attack#Mathematics).

So, for example, with 64-bit IDs, then we need ~190M of IDs to find a collision with probability 0.1%. 

What's an acceptable number? I think that 128 bits are more than enough (in comparison, secp256k1 provides ~128 bits of security), but I guess something in the range between 64 and 128 is good.

Reply:
Yeah, I typically glance at the wikipedia article whenever I'm in this situation (actually, I find [this very similar one](https://en.wikipedia.org/wiki/Birthday_problem#Probability_table) to be a bit more instructive), but what I don't have (and can't immediately find with a bit of googling) is a practical, situational guide about applying birthday problem math to situations like this.

You're saying between 64 and 128 is good (surely closer to 64, right?), but what makes you say that this is in the 64-128 range rather than, say, 32 bits?  When you are playing out a scenario, how do you know how to think about values for p and the collision likelihood?

Reply:
I don't think, human readability is really a requirement.
If one _really_ wants human readability, a better way would be to combine with something like ENS

Reply:
I think human readability is important; this is one of the most important pieces of data appearing in logs, and one of the only low-sensitivity bytestrings appearing in log emissions for all three Characters.